test/RibbonTreasuryVaultWithAirSwap.ts

```diff
@@ -0,0 +1,3679 @@
+            ).strikePrice()
+          ).toString(),
+          newStrikePrice.toString()
+        );
+
+        const expiryTimestampOfNewOption = await (
+          await getContractAt("IOtoken", await vault.nextOption())
+        ).expiryTimestamp();
+
+        assert.bnEqual(
+          await vault.currentOtokenPremium(),
+          (
+            await optionsPremiumPricer.getPremiumInStables(
+              newStrikePrice,
+              expiryTimestampOfNewOption,
+              params.isPut
+            )
+          )
+            .mul(await vault.premiumDiscount())
+            .div(1000)
+        );
+      });
+
+      it("closes short even when otokens are burned", async function () {
+        await approve(assetContract, vault, depositAmount, userSigner);
+        await depositIntoVault(collateralAsset, vault, depositAmount);
+
+        await rollToNextOption();
+
+        // we couldn't get any bids OTC, so we just burn the minted oTokens
+        // and return 100% of the tokens
+
+        await vault.connect(keeperSigner).burnRemainingOTokens();
+
+        await rollToSecondOption(firstOption.strikePrice);
+
+        const controller = await ethers.getContractAt(
+          "IController",
+          GAMMA_CONTROLLER[chainId]
+        );
+
+        assert.equal(await controller.getAccountVaultCounter(vault.address), 2);
+      });
+
+      it("closes short when otokens are partially burned", async function () {
+        await approve(assetContract, vault, depositAmount, userSigner);
+        await depositIntoVault(collateralAsset, vault, depositAmount);
+
+        await rollToNextOption();
+
+        const otoken = await ethers.getContractAt(
+          "IERC20",
+          firstOption.address
+        );
+
+        // Purchase half the oTokens
+        const initialOtokenBalance = await otoken.balanceOf(vault.address);
+        const totalOptionsAvailableToBuy = initialOtokenBalance.div(2);
+        let decimals = premiumInStables ? premiumDecimals : tokenDecimals;
+        const bid = wmul(
+          totalOptionsAvailableToBuy.mul(BigNumber.from(10).pow(10)),
+          firstOptionPremium
+        )
+          .div(BigNumber.from(10).pow(18 - decimals))
+          .toString();
+
+        // Assume premium is in stables
+        const signedOrder = await signOrderForAirSwap({
+          vaultAddress: vault.address,
+          counterpartyAddress: userSigner.address,
+          signerPrivateKey: userSignerPrivateKey,
+          sellToken: otoken.address,
+          buyToken: premiumContract.address,
+          sellAmount: totalOptionsAvailableToBuy.toString(),
+          buyAmount: bid,
+        });
+
+        await premiumContract
+          .connect(userSigner)
+          .approve(airswapContract.address, bid);
+
+        await vault.connect(keeperSigner).sellOptions(signedOrder);
+
+        // since some oTokens sold, we distribute the premiums
+        await vault.connect(keeperSigner).chargeAndDistribute();
+
```

Comment:
<img width="895" alt="image" src="https://user-images.githubusercontent.com/99089658/210387537-f51224fe-52ad-417e-a2e3-57916fac4c56.png">

I'd say we can keep a similar test as in the original file here:

eg.
        assert.bnEqual(
          await otoken.balanceOf(vault.address),
          initialOtokenBalance.div(2)
        );

Reply:
Addressed in [8928237](https://github.com/ribbon-finance/ribbon-v2/pull/297/commits/892823778c487e6773542058138ce27725910b5a)